pymatgen/analysis/local_env.py

```diff
@@ -641,7 +640,62 @@ def get_voronoi_polyhedra(self, structure, n):
         # Extract data about the site in question
         return self._extract_cell_info(structure, 0, neighbors, targets, voro)
 
+    def get_all_voronoi_polyhedra(self, structure):
+        """Get the Voronoi polyhedra for all site in a simulation cell
+
+        Args:
+            structure (Structure): Structure to be evaluated
+        Returns:
+            A dict of sites sharing a common Voronoi facet with the site
+            n mapped to a directory containing statistics about the facet:
+                - solid_angle - Solid angle subtended by face
+                - angle_normalized - Solid angle normalized such that the
+                    faces with the largest
+                - area - Area of the facet
+                - face_dist - Distance between site n and the facet
+                - volume - Volume of Voronoi cell for this face
+                - n_verts - Number of vertices on the facet
+        """
+
+        # Assemble the list of neighbors used in the tessellation
+        if self.targets is None:
+            targets = structure.composition.elements
+        else:
+            targets = self.targets
+
+        # Populate an initial list of sites and their indices (id + supercell)
+        sites = list(structure)
+        indices = [(i, 0, 0, 0) for i in range(len(sites))]
+
+        # Get all neighbors within a certain cutoff
+        #   Record both the list of these neighbors, and the site indices
+        all_neighs = structure.get_all_neighbors(self.cutoff, include_index=True)
+        for neighs in all_neighs:
+            sites.extend([x[0] for x in neighs])
+            indices.extend([(x[2],) + NearNeighbors._get_image(x[0].frac_coords) for x in neighs])
+
+        # Get the non-duplicates (using the site indices for performance)
+        _, uniq_indices = np.unique(indices, return_index=True, axis=0)
+        sites = np.array(sites)[sorted(uniq_indices)]
+        del indices  # Save memory (tessellations can be costly)
+
+        # Run the tessellation
+        qvoronoi_input = [s.coords for s in sites]
+        voro = Voronoi(qvoronoi_input)
+
+        # Get the information for each neighbor
+        return [self._extract_cell_info(structure, i, sites, targets, voro)
+                for i in range(len(structure))]
+
     def _get_elements(self, site):
```

Comment:
Also, the `_get_elements` method seems out of place here as it seems like this would be useful for a number of other operations. If you agree, should I open a separate PR to move it to Site?